test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +79,77 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-unanalyzed-fields [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (assert (pos? ;; don't let ourselves be fooled if the test passes because the table is
+                  ;; totally broken or has no fields. Make sure we actually test something
+             (db/count Field :table_id table-id)))
+    (db/count Field :last_analyzed nil :table_id table-id)))
+
+(defn latest-sync-time [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (db/select-field :last_analyzed Field :table_id table-id)))
```

Comment:
although that is still sort of a lie. It's not necessarily the last sync time. Instead you would want to do

```clojure
(db/select-one-field :last_analyzed 'Field
   :last_analyzed [:not= nil] 
  :table_id <id>
  {:order-by [[:last_analyzed :desc]]})
```

But that's getting complicated since you don't actually use the date, right?

You might just want to change this function to something like `has-been-synced?` and then change the db call to

```clojure
(db/exists? Field, :last_analyzed [:not= nil], :table_id <id>)
```